packages/core/frontend/sirius-components-omnibox/src/Omnibox.tsx

```diff
@@ -0,0 +1,179 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { IconOverlay, useSelection } from '@eclipse-sirius/sirius-components-core';
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import Breadcrumbs from '@mui/material/Breadcrumbs';
+import Dialog from '@mui/material/Dialog';
+import Divider from '@mui/material/Divider';
+import FormControl from '@mui/material/FormControl';
+import Input from '@mui/material/Input';
+import List from '@mui/material/List';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import Typography from '@mui/material/Typography';
+import { useEffect, useRef, useState } from 'react';
+import { makeStyles } from 'tss-react/mui';
+import { OmniboxAction, OmniboxProps, OmniboxState } from './Omnibox.types';
+import { useOmniboxCommands } from './useOmniboxCommands';
+import { GQLGetOmniboxCommandsQueryVariables } from './useOmniboxCommands.types';
+
+const useOmniboxStyles = makeStyles()((theme) => ({
+  omnibox: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  omniboxInputArea: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  omniboxIcon: {
+    minWidth: theme.spacing(4),
+    fontSize: '2.5rem',
+  },
+  omniboxFormControl: {
+    paddingTop: theme.spacing(1),
+    paddingBottom: theme.spacing(1),
+    paddingLeft: theme.spacing(0.5),
+    paddingRight: theme.spacing(0.5),
+  },
+  omniboxResultArea: {
+    overflowY: 'auto',
+  },
+}));
+
+export const Omnibox = ({ open, initialContextEntries, onClose }: OmniboxProps) => {
+  const [state, setState] = useState<OmniboxState>({
+    contextEntries: initialContextEntries,
+    query: '',
+    actions: [],
+  });
+
+  const { setSelection } = useSelection();
+  const { getOmniboxCommands, loading, data } = useOmniboxCommands();
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  const listRef = useRef<HTMLUListElement>(null);
+
+  const handleChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement> = (event) => {
+    const {
+      target: { value },
+    } = event;
+    setState((prevState) => ({ ...prevState, query: value }));
+  };
+
+  useEffect(() => {
+    const variables: GQLGetOmniboxCommandsQueryVariables = {
+      contextEntries: state.contextEntries,
+      query: state.query,
+    };
+    getOmniboxCommands({ variables });
+  }, [state.query]);
```

Comment:
NO! Do not perform the query, and thus `eAllContents` each time the query changes. Wait for the user to type "enter"

Reply:
If the user has to hit "enter" to send the query, how does he validate the option from the results he wants?